### ISSUE TITLE
Provide better and more thumbnail images by default

### ISSUE THREAD

**Issue by LukasKalbertodt** [opencast/opencast#3753]
Many Opencast-related applications require "thumbnail images" for videos. They are used for a quick preview when multiple videos are shown in a list or grid. The current default workflows generate two image attachments for this general purpose:

- `*/player+preview`: 720px high
- `*/search+preview`: 160px wide

The intention is good, but I think we have two problems here: (a)  for most "normal thumbnail usages", both of these sizes are not great, and (b) there should be more than two size options. 

Let's take a look at some thumbnails in Opencast-related apps! One important thing to consider: by now, many people are using some form of UI scaling by their OS as pixel density has increased a lot in the recent years. I am at 200% scaling on my 4k screen all the time. So the pixel dimensions of images reported by the devtools in the browser are not the physical resolution! I will keep the physical resolution for 200% scaling in parenthesis.

- Vallery: 160x90 (320x180)
- Tobira Desktop: 240x135 (480x270)
- Stud.IP plugin: 330x185 (660x370)
- Tobira Mobile: 360x202 (720x404)
- YouTube (for comparison) on my desktop: 360x202 (720x404)

So the only case where the default `search+preview` would look sharp is for Vallery on 100% scaling. For all other cases, it results in a blurry thumbnail, which is really not great for presenting videos and just makes everything look bad.

<details>
<summary>Example of using `search+preview` and `player+preview` in Tobira</summary>

(Of course, this is also stretched here in GitHub, but you get the point)

**Bad**
![image](https://user-images.githubusercontent.com/7419664/167845650-4da51a53-c46f-4322-80ab-3f7fb20245ad.png)

**Good**
![image](https://user-images.githubusercontent.com/7419664/167848743-7df2b346-23d3-49ff-b890-4bd213f67110.png)

</details>

That's why Tobira and Stud.IP already use the `player+preview` image, despite it being a lot too big. At least it looks good, but it wastes bandwidth, which is especially bad for mobile users. As we can see, there is no good default thumbnail an external application can use.

Regarding point (b), notice how I listed "Tobira desktop" and "Tobira mobile" up there. The size of thumbnails in apps often change, depending on the screen size. Also, users have different scale factors, so one image is not the best choice for all. It is possible to define a list of images of different sizes and let the browser decide: this is the best case. However, for that, we obviously need multiple images.

---

With this long motivation out of the way, I suggest to change the default workflows to generate a number of thumbnail images with different sizes. I am thinking between 6 and 12 images. Just to throw one example of image widths out there: 120, 160, 220, 300, 400, 520, 660, 820, 1000. Obviously I'm not married to any particular image size. 

This leaves the question of identifying the attachments. I'm not sure if the size should be in the flavor? `*/thumbnail-520`? Or should we put it in the attachment `additionalProperties`, like `timeline-preview` does it. Anyway, I don't have any opinion about this.

I think ffmpeg should be able to extract all image sizes with one command, so I doubt this would slow down workflows in any significant way.

**Comment by mtneug** [comment]
Has this been discussed further?

**Comment by LukasKalbertodt** [comment]
I don't think so. Maybe something for the next technical meeting?

**Comment by LukasKalbertodt** [comment]
Additional idea: Opencast should also generate an extremely low res version blurred of the thumbnail. This can then be included (base64 encoded) in API responses directly, making it possible to show a nicer placeholder while the image loads. I haven't read up on the details here, but I'm sure all of us saw this effect already: an extremely blurred version of an image that is about to be loaded.

**Comment by mtneug** [comment]
> Additional idea: Opencast should also generate an extremely low res version blurred of the thumbnail. This can then be included (base64 encoded) in API responses directly, making it possible to show a nicer placeholder while the image loads. I haven't read up on the details here, but I'm sure all of us saw this effect already: an extremely blurred version of an image that is about to be loaded.

Instead of this, I would suggest first experimenting with progressive JPEGs or equivalent formats. This feature is thus basically built in.

**Comment by LukasKalbertodt** [comment]
> Instead of this, I would suggest first experimenting with progressive JPEGs or equivalent formats. This feature is thus basically built in.

Yes and no. I'm definitely in favor of experimenting and using progressive images. Newer formats also have nicer behavior for this feature. But all of that still requires sending an additional HTTP request and receiving at least a few bytes there. The point of the other solution is to include the few pixels in the API, not requiring any additional HTTP request to show the placeholder. And the additional size in the API request is not that much, very much comparable to other metadata fields. 

To be clear, I'm not saying this is super important, but if someone tackles this issue, and if it's easy to include, I'd be in favor.
